src/Protocol/Protocol_1_14.cpp

```diff
@@ -215,77 +215,77 @@ cProtocol::Version cProtocol_1_14::GetProtocolVersion()
 
 
 
-bool cProtocol_1_14::HandlePacket(cByteBuffer & a_ByteBuffer, UInt32 a_PacketType)
+std::pair<short, short> cProtocol_1_14::GetItemFromProtocolID(UInt32 a_ProtocolID)
 {
-	if (m_State != 3)
-	{
-		return Super::HandlePacket(a_ByteBuffer, a_PacketType);
-	}
-
-	// Game
-	switch (a_PacketType)
-	{
-		default: AString dum; a_ByteBuffer.ReadAll(dum); a_ByteBuffer.CommitRead(); a_ByteBuffer.Write(" ", 1);
-			return true;
-	}
+	return PaletteUpgrade::ToItem(Palette_1_14::ToItem(a_ProtocolID));
 }
 
 
 
 
 
-void cProtocol_1_14::HandlePacketBlockDig(cByteBuffer & a_ByteBuffer)
+UInt32 cProtocol_1_14::GetProtocolBlockType(BLOCKTYPE a_BlockType, NIBBLETYPE a_Meta)
 {
+	return Palette_1_14::From(PaletteUpgrade::FromBlock(a_BlockType, a_Meta));
 }
 
 
 
 
 
-void cProtocol_1_14::HandlePacketBlockPlace(cByteBuffer & a_ByteBuffer)
+UInt32 cProtocol_1_14::GetProtocolItemType(short a_ItemID, short a_ItemDamage)
 {
+	return Palette_1_14::From(PaletteUpgrade::FromItem(a_ItemID, a_ItemDamage));
 }
 
 
 
 
 
-void cProtocol_1_14::HandlePacketUpdateSign(cByteBuffer & a_ByteBuffer)
+UInt32 cProtocol_1_14::GetProtocolStatisticType(Statistic a_Statistic)
 {
+	return Palette_1_14::From(a_Statistic);
 }
 
 
 
 
 
-std::pair<short, short> cProtocol_1_14::GetItemFromProtocolID(UInt32 a_ProtocolID)
+bool cProtocol_1_14::HandlePacket(cByteBuffer & a_ByteBuffer, UInt32 a_PacketType)
 {
-	return PaletteUpgrade::ToItem(Palette_1_14::ToItem(a_ProtocolID));
+	if (m_State != 3)
```

Comment:
Magic number

Reply:
He's just alpha-sorting the functions I think. That magic number was always there.

Reply:
muahaha! if you think i'm only alpha sorting then my EVIL PLAN has succeeded!!1!

anyway I can also make it an enum, it's a cleanup anyway.

Reply:
made it a plain enum. Scoped enum will be a mass-replace later due to all the assert(m_State == 3) everywhere.